bagel/pheno_utils.py

```diff
@@ -42,9 +44,21 @@ def validate_portal_uri(portal: str) -> Optional[str]:
 def load_pheno(input_p: Path) -> pd.DataFrame | None:
     """Load a .tsv pheno file and do some basic validation."""
     if input_p.suffix == ".tsv":
-        pheno_df: DataFrame = pd.read_csv(
-            input_p, sep="\t", keep_default_na=False, dtype=str
-        )
+        try:
+            pheno_df: DataFrame = pd.read_csv(
+                input_p,
+                sep="\t",
+                keep_default_na=False,
+                dtype=str,
+                encoding="utf-8",
+            )
+        except UnicodeDecodeError as e:
+            # TODO: Refactor once https://github.com/neurobagel/bagel-cli/issues/218 is addressed
+            typer.echo(
+                file_encoding_error_message(input_p),
+                err=True,
+            )
+            raise typer.Exit(code=1) from e
```

Comment:
Very cool. Just out of curiosity: [rich](https://rich.readthedocs.io/en/stable/index.html) (which does the coloring of the terminal for typer afaik) does not have a way to color / make nice custom error classes? I am mainly asking because from reading (not trying) this I assume our pretty colorful text would be immediately followed by the stacktrace from hell that typer already generates, right?

Reply:
Actually, using typer.echo + typer.Exit like this hides the stacktrace entirely :) Only our pretty message is shown in this case.

So as part of https://github.com/neurobagel/bagel-cli/issues/218, we can refactor other error messages using this approach, to hide the stacktrace when it is not necessary for input validation. In comparison, elsewhere in the code where we're currently using a simple `raise` in an Except block to output a custom message, the stacktrace is shown and our custom error message appears at the very end.

And yes, I believe you can achieve the same custom message styling using `rich` [print](https://typer.tiangolo.com/tutorial/printing/#use-rich-print) instead of `typer.echo` - I just used the simpler typer.echo to start!